app/controllers/application_controller.rb

```diff
@@ -1,3 +1,9 @@
 class ApplicationController < ActionController::Base
   protect_from_forgery with: :exception
+  before_action :api_request
+
+  def api_request
```

Comment:
We typically don't use before_actions for the main logic in controllers but rather for operations before getting to the main logic (i.e. making sure a user is logged in before making an api call or querying the database)